src/maxwell-web-view.c

```diff
@@ -801,6 +680,109 @@ maxwell_web_view_draw (GtkWidget *widget, cairo_t *cr)
   return FALSE;
 }
 
+static void
+child_update_visibility (MaxwellWebView *webview, GtkWidget *child)
+{
+  MaxwellWebViewPrivate *priv = MAXWELL_WEB_VIEW_PRIVATE (webview);
+  ChildData *data = get_child_data_by_child (priv, child);
+
+  if (priv->script_loaded && data && gtk_widget_get_name (data->child))
+    js_run_printf (webview, "maxwell.child_set_visible ('%s', %s);",
+                   gtk_widget_get_name (data->child),
+                   gtk_widget_get_visible (child) ? "true" : "false");
+}
+
+static void
+on_child_visible_notify (GObject        *object,
+                         GParamSpec     *pspec,
+                         MaxwellWebView *webview)
+{
+  MaxwellWebViewPrivate *priv = MAXWELL_WEB_VIEW_PRIVATE (webview);
+
+  if (priv->script_loaded)
+    child_update_visibility (webview, GTK_WIDGET (object));
```

Comment:
This function is already robust to `!priv->script_loaded`, so you don't need the extra check here.